### PR TITLE
Simplify init function of `DataStore` class

### PR DESCRIPTION
**Pull request by lmohrmann**
As discussed in #1790, I've removed the `from_files` function and extended `from_dir` such that filenames can be specified explicitly if desired. Comments welcome!

### CODE REVIEW

File: gammapy/data/data_store.py [hunk 1]
```diff
@@ -51,37 +51,41 @@ def __str__(self):
...
     @classmethod
-    def from_files(cls, base_dir, hdu_table_filename=None, obs_table_filename=None):
-        """Construct from HDU and observation index table files."""
-        if hdu_table_filename:
-            log.debug("Reading {}".format(hdu_table_filename))
-            hdu_table = HDUIndexTable.read(str(hdu_table_filename), format="fits")
+    def from_dir(cls, base_dir, hdu_table_filename=None, obs_table_filename=None):
+        """Create from a directory.
+
+        If no filenames are specified, files with default filenames
```
Comment (adonath) [inline gammapy/data/data_store.py]: Could you a `Parameters` section here instead, that describes the method arguments, like we do for every method in Gammapy? You will find a lot of examples how to do it in the same source file.

**Comment by lmohrmann** [comment]
@adonath I've fixed the docstring and added a test (not sure this is how you meant it?)

**Comment by cdeil** [comment]
I'm merging this now, because I want to continue with the update of tests that use HESS data files started in #1796 , and the remaining one is for datastore / index files and I want to avoid merge conflicts.

@adonath - Feel free to adjust directly in master as you like if there's something that should be changed.

@lmohrmann - Thank you!

**Comment by adonath** [comment]
Looks good to me, thanks @lmohrmann!

**Pull request merged**